src/main/java/fr/sncf/osrd/infra_state/routes/ControlledRouteState.java

```diff
@@ -0,0 +1,174 @@
+            if (!tvdSectionsBehind.contains(releaseGroup))
+                continue;
+            for (var releasedTvdSection : releaseGroup) {
+                var tvdSection = sim.infraState.getTvdSectionState(releasedTvdSection.index);
+                if (tvdSection.isReserved())
+                    tvdSection.free(sim);
+            }
+        }
+    }
+
+    /** Notify the route that one of his tvd section is reserved */
+    @Override
+    public void onTvdSectionReserved(Simulation sim) throws SimulationError {
+        if (status != FREE)
+            return;
+        updateStatus(sim, CONFLICT);
+    }
+
+    /** Notify the route that one of his tvd section is occupied */
+    @Override
+    public void onTvdSectionOccupied(Simulation sim, TVDSection tvdSection) throws SimulationError {
+
+        if (status == REQUESTED || status == FREE) {
+            throw new SimulationError("The TVD section we try to occupy isn't reserved yet");
+        }
+        if (status != RouteStatus.RESERVED)
+            return;
+
+        updateStatus(sim, OCCUPIED);
+    }
+
+    /**
+     * Request all switches to move to the position defined by
+     * route.switchesPosition
+     *
+     * @param sim the current simulation
+     */
+    private void requestSwitchPositionChange(Simulation sim) throws SimulationError {
+        // Set the switches in the moving position
+        movingSwitchesLeft = 0;
+        for (var switchGrp : route.switchesGroup.entrySet()) {
+            var switchRef = switchGrp.getKey();
+            var group = switchGrp.getValue();
+            var switchState = sim.infraState.getSwitchState(switchRef.switchIndex);
+            boolean isActiveGrp = switchState.getGroup().equals(group);
+            if (!isActiveGrp) {
+                movingSwitchesLeft++;
+                switchState.requestGroupChange(sim, group, this);
+            }
+        }
+    }
+
+    /**
+     * Reserve the tvd sections of the route
+     *
+     * @param sim the current simulation
+     */
+    public void reserveTvdSection(Simulation sim) throws SimulationError {
+        for (var tvdSectionPath : route.tvdSectionsPaths) {
+            var tvdSection = sim.infraState.getTvdSectionState(tvdSectionPath.tvdSection.index);
+            tvdSection.reserve(sim);
+        }
+    }
+
+    /**
+     * Reserve a route and his tvd sections, in CBTC mode if specified. Routes that share tvd sections will
+     * have the status CONFLICT
+     */
+    @Override
+    protected void reserveWithGivenCBTC(Simulation sim, boolean cbtc) throws SimulationError {
+        if (!(status == FREE || (cbtc && hasCBTCStatus())))
+            throw new SimulationError("The route we try to reserve isn't free");
+
+        isCBTCReserved = cbtc;
+
+        requestSwitchPositionChange(sim);
+
+        RouteStatus newStatus = movingSwitchesLeft > 0 ? RouteStatus.REQUESTED : RouteStatus.RESERVED;
+
+        updateStatus(sim, newStatus);
+
+        reserveTvdSection(sim);
```

Comment:
I find that the line jumps is not very elegant... 
`isCBTCReserved` should be modified in a change